src/gameLogic/generation/actions/gadget/FogTin.cpp

```diff
@@ -3,12 +3,39 @@
 //
 
 #include "gameLogic/generation/ActionGenerator.hpp"
+#include "gameLogic/validation/ActionValidator.hpp"
+#include "util/GameLogicUtils.hpp"
+#include "datatypes/gameplay/GadgetAction.hpp"
 
 namespace spy::gameplay {
     std::vector<std::shared_ptr<BaseOperation>>
-    ActionGenerator::generateFogTin(const State &/*s*/, const util::UUID &/*activeCharacter*/) {
+    ActionGenerator::generateFogTin(const State &s, const util::UUID &activeCharacter,
+                                    const spy::MatchConfig &config) {
 
-        // TODO implement
-        return {nullptr};
+        std::vector<std::shared_ptr<BaseOperation>> valid_ops;
+        std::vector<util::Point> allPoints;
+
+        auto character = s.getCharacters().findByUUID(activeCharacter);
+        for (unsigned int range = 2; range < config.getFogTinRange(); range++) {
```

Comment:
Warum fängt man bei einer range von 2 an?

Reply:
Damit man sich nicht selbst in den Nebel setzen kann